### ISSUE TITLE
[RESEARCH] CS/keyword: webhook

### ISSUE THREAD

**Issue by jakesung-harborx** [asunlabs/mycodebox-monorepo#66]
# topic : understanding the definition of webhook in computer science
## read this 

> A webhook is an HTTP-based callback function that allows lightweight, event-driven communication between 2 [application programming interfaces (APIs)](https://www.redhat.com/en/topics/api/what-are-application-programming-interfaces). Webhooks are used by a wide variety of web apps to receive small amounts of data from other apps, but webhooks can also be used to trigger automation workflows in [GitOps](https://www.redhat.com/en/topics/devops/what-is-gitops) environments.

> To set up a webhook, the client gives a unique URL to the server API and specifies which event it wants to know about. Once the webhook is set up, the client no longer needs to poll the server; the server will automatically send the relevant payload to the client’s webhook URL when the specified event occurs. 

> Webhooks are often referred to as reverse APIs or push APIs, because they put the responsibility of communication on the server, rather than the client. Instead of the client sending HTTP requests—asking for data until the server responds—the server sends the client a single HTTP POST request as soon as the data is available. Despite their nicknames, webhooks are not APIs; they work together. An application must have an API to use a webhook. 

> The name webhook is a simple combination of web, referring to its HTTP-based communication, and the hooking programming function that allows apps to intercept calls or other events that might be of interest. 

## reference

- https://www.redhat.com/en/topics/automation/what-is-a-webhook

**Comment by developerasun** [comment]
## Webhook in plain English

> Client -> Server: Heres my credit card number
Server -> Client: Thanks! Let me process that.

> Server -> External API: Please process this credit card for the amount of $X and ping me at this webhook when you're done.
External API -> Server: OK, will let you know when I'm done.

> Client -> Server: You done yet?
Server -> Client: Nope!

> External API -> Server (to webhook): Hey, I'm done, here's the receipt.
Server -> External API: Thanks!

> Client -> Server: You done yet?
Server -> Client: Yes! Here's your receipt.

## Reference 

- https://stackoverflow.com/questions/69148643/how-does-one-create-a-webhook-for-a-personal-project-javascript-node
